strawberry/type.py

```diff
@@ -118,6 +127,66 @@ class StrawberryOptional(StrawberryContainer):
     ...
 
 
+class StrawberryAnnotated(StrawberryContainer):
+    """
+    Equivalent to typing.Annotated for strawberry types
+    """
+
+    def __init__(self, of_type: Union[StrawberryType, type], *args):
+        of_type, base_args = StrawberryAnnotated.get_type_and_args(of_type)
+        super().__init__(of_type)
+        self.args = base_args + list(args)
+
+    def __eq__(self, other: object) -> bool:
+        if isinstance(other, StrawberryType):
+            if type(self) == type(other):
```

Comment:
Why not `isinstance(other, type(self))`?
This would also be resilient if we ever have a subclass for this type.